internal/service/dynamodb/global_secondary_index.go

```diff
@@ -0,0 +1,710 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package dynamodb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	awstypes "github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/dlclark/regexp2"
+	"github.com/hashicorp/aws-sdk-go-base/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/enum"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	"github.com/hashicorp/terraform-provider-aws/internal/flex"
+	"github.com/hashicorp/terraform-provider-aws/internal/slices"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+const (
+	resNameGlobalSecondaryIndex = "global_secondary_index"
+)
+
+// @SDKResource("aws_dynamodb_global_secondary_index", name="Global Secondary Index")
```

Comment:
All new resource types must be implemented using [Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework) instead of [Plugin SDK v2](https://developer.hashicorp.com/terraform/plugin/sdkv2).

The resource implementation can be partially converted using the tool `tfsdk2fw` found at https://github.com/hashicorp/terraform-provider-aws/tree/main/tools/tfsdk2fw